src/Module/AbstractBaseModule.php

```diff
@@ -187,26 +209,57 @@
      */
     protected $compContainerFactory;
 
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|ArrayAccess|ContainerInterface $config               The module config.
+     * @param ConfigFactoryInterface                        $configFactory        The config factory.
+     * @param ContainerFactoryInterface                     $containerFactory     The container factory.
+     * @param ContainerFactoryInterface                     $compContainerFactory The composite container factory.
+     * @param EventManagerInterface|null                    $eventManager         The event manager, or null.
+     * @param EventFactoryInterface|null                    $eventFactory         The event factory, or null.
+     */
+    public function __construct(
+        $config,
+        ConfigFactoryInterface $configFactory,
+        ContainerFactoryInterface $containerFactory,
+        ContainerFactoryInterface $compContainerFactory,
+        EventManagerInterface $eventManager = null,
+        EventFactoryInterface $eventFactory = null
+    ) {
+        $this->_initModule($config, $configFactory, $containerFactory, $compContainerFactory);
+        $this->_initModuleEvents($eventManager, $eventFactory);
+    }
+
     /**
      * Initializes the module with all required information.
      *
      * @since [*next-version*]
      *
-     * @param string|Stringable         $key                  The module key.
-     * @param string[]|Stringable[]     $dependencies         The module dependencies.
-     * @param ConfigFactoryInterface    $configFactory        The config factory.
-     * @param ContainerFactoryInterface $containerFactory     The container factory.
-     * @param ContainerFactoryInterface $compContainerFactory The composite container factory.
+     * @param array|stdClass|ArrayAccess|ContainerInterface $config               The module config.
+     * @param ConfigFactoryInterface                        $configFactory        The config factory.
+     * @param ContainerFactoryInterface                     $containerFactory     The container factory.
+     * @param ContainerFactoryInterface                     $compContainerFactory The composite container factory.
      */
     protected function _initModule(
-        $key,
-        $dependencies,
+        $config,
         ConfigFactoryInterface $configFactory,
         ContainerFactoryInterface $containerFactory,
         ContainerFactoryInterface $compContainerFactory
     ) {
-        $this->_setKey($key);
-        $this->_setDependencies($dependencies);
+        $config = $configFactory->make([
+            ConfigFactoryInterface::K_DATA => $config,
```

Comment:
Doesn't [look](https://github.com/Dhii/map/blob/develop/src/MakeCapableMapTrait.php#L44) like `ContainerInterface` is allowed here.

Reply:
[Looks like it is](https://github.com/Dhii/map/blob/develop/src/MakeCapableMapTrait.php#L29). If it shouldn't be, and this is a case of misdocumentation, I will amend. But confirmation would be appreciated.

Reply:
Yep, looks like the doc is wrong. Clearly, `_normalizeIterable()` would fail on a `ContainerInterface` alone.